common4j/src/main/com/microsoft/identity/common/java/commands/parameters/nativeauth/AcquireTokenNoFixedScopesCommandParameters.java

```diff
@@ -0,0 +1,156 @@
+import com.microsoft.identity.common.java.exception.TerminalException;
+import com.microsoft.identity.common.java.logging.Logger;
+import com.microsoft.identity.common.java.providers.microsoft.azureactivedirectory.AzureActiveDirectory;
+import com.microsoft.identity.common.java.providers.microsoft.azureactivedirectory.AzureActiveDirectoryCloud;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+
+@Getter
+@EqualsAndHashCode(callSuper = true)
+@SuperBuilder(toBuilder = true)
+public class AcquireTokenNoFixedScopesCommandParameters extends BaseNativeAuthCommandParameters {
+
+    private static final String TAG = AcquireTokenNoFixedScopesCommandParameters.class.getSimpleName();
+
+    private static final Object sLock = new Object();
+
+    private final IAccountRecord account;
+
+    @Expose()
+    private final Set<String> scopes;
+
+    @Expose()
+    private final String claimsRequestJson;
+
+    @Expose()
+    private final AbstractAuthenticationScheme authenticationScheme;
+
+    @Expose()
+    private final String mamEnrollmentId;
+
+    @Expose()
+    private final boolean forceRefresh;
+
+    private final String loginHint;
+
+    private final List<Map.Entry<String, String>> extraOptions;
+
+    public Set<String> getScopes() {
+        return this.scopes == null ? null : new HashSet<>(this.scopes);
+    }
+
+    public String getMamEnrollmentId(){
+        return mamEnrollmentId;
+    }
+
+    public void validate() throws ArgumentException {
+        final String methodName = ":validate";
+
+        Logger.verbose(
+                TAG + methodName,
+                "Validating operation params..."
+        );
+
+        // AuthenticationScheme is present...
+        if (null == authenticationScheme) {
+            throw new ArgumentException(
+                    ArgumentException.ACQUIRE_TOKEN_SILENT_OPERATION_NAME, // TODO update argument name
+                    ArgumentException.AUTHENTICATION_SCHEME_ARGUMENT_NAME,
+                    "authentication scheme is undefined"
+            );
+        }
+
+        if (getAccount() == null) {
+            Logger.warn(TAG, "The account set on silent operation parameters is NULL.");
+            // if the authority is B2C, then we do not need check if matches with the account environment
+            // as B2C only exists in one cloud and can use custom domains
+            // This logic should also apply to CIAM authorities
+        }
+    }
+
+    /**
+     * Note - this method may throw a variety of RuntimeException if we cannot perform cloud
+     * discovery to determine the set of cloud aliases.
+     * @return true if the authority matches the cloud environment that the account is homed in.
+     */
+    private boolean authorityMatchesAccountEnvironment() {
```

Comment:
Do you plan to support multiples environment for Native Auth?

Reply:
Actually what clouds today support native auth?

Reply:
If you really need to do this check then why not just re-use the code?

Reply:
We support public cloud for now.

Reply:
Ok, but you plan to support other clouds in the future?

And you want to keep this check in the code just in case someone sends you wrong authority (wrong cloud)?

Reply:
This is unused. I have removed it.